back/app/controllers/web_api/v1/cosponsorships_controller.rb

```diff
@@ -19,7 +19,7 @@ def index
   def accept_cosponsorship
     if @cosponsorship.status == 'pending'
       @cosponsorship.update(status: 'accepted')
```

Comment:
Aware this is from the previous PR, but I missed it there. There is a check missing on whether the `update` operation succeeds. then, the else branch of _that_ check should call `render: json: { errors:....`. Currently, there will be no error for the status returned and if the `update` fails (that can be for any reason, maybe future code changes or the data becoming invalid in some way) there will be no error message returned.

An elegant way of solving this is to move the `status == pending` check to the model, but make it a conditional validation that is only called when doing this accept. By doing so, it ends up in your model errors and you only need to check whether the  `update` operations succeeds. See for example 
[this cnotroller](https://github.com/CitizenLabDotCo/citizenlab/blob/1c214c12f68cc4a53842fefdd3b40cc852bdd7a9/back/app/controllers/web_api/v1/invites_controller.rb#L138) in combination with [this model validation](https://github.com/CitizenLabDotCo/citizenlab/blob/1c214c12f68cc4a53842fefdd3b40cc852bdd7a9/back/app/models/user.rb#L183-L184)

Reply:
It was taking a bit of time to figure out this setup so I added the check directly into the controller as I've seen we do the vast majority of cases. Feel free to modify it in your PR, though, if you feel this is not good enough. :)